File: code/scenes/src/main/java/org/fbme/scenes/controllers/scene/SceneStateKey.java
package org.fbme.scenes.controllers.scene;

import org.jetbrains.annotations.NotNull;

public class SceneStateKey<T> {

    @NotNull
    private final String myId;

    public SceneStateKey(@NotNull String id) {
        this.myId = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!((o instanceof SceneStateKey))) {
            return false;
        }

        SceneStateKey that = (SceneStateKey) o;
        return myId.equals(that.myId);
    }

    @Override
    public int hashCode() {
        return myId.hashCode();
    }

    @Override
    public String toString() {
        return "SceneStateKey{" + myId + "}";
    }
}

File: code/scenes/src/main/kotlin/org/fbme/scenes/controllers/scene/SceneStateKey.kt
package org.fbme.scenes.controllers.scene

class SceneStateKey<T>(private val id: String) {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other !is SceneStateKey<*>) {
            return false
        }
        return id == other.id
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }

    override fun toString(): String {
        return "SceneStateKey{$id}"
    }
}
